docs/security.md

```diff
@@ -31,6 +31,7 @@ SSL must be configured on each node and configured for each component involved i
 
 ### YARN mode
 The key-store can be prepared on the client side and then distributed and used by the executors as the part of the application. It is possible because the user is able to deploy files before the application is started in YARN by using `spark.yarn.dist.files` or `spark.yarn.dist.archives` configuration settings. The responsibility for encryption of transferring these files is on YARN side and has nothing to do with Spark.
+For long-running apps like Spark Streaming apps be able to write to HDFS, it is possible to pass a principal and keytab to `spark-submit` via the `--principal` and `--keytab` parameters respectively. The keytab passed in will be copied over to the machine running the Application Master securely via the Hadoop Distributed Cache. The Kerberos login will be periodically renewed using this principal and keytab and the delegation tokens required for HDFS will be generated periodically so the application can continue writing to HDFS.
```

Comment:
"securely via the Hadoop Distributed Cache" -> maybe qualify that it's only secure if YARN is configured to use SSL and HDFS is set up to use encryption?

Also, could you break this line? Long lines suck for editing.